packages/twenty-front/src/modules/ui/feedback/snack-bar-manager/hooks/useSnackBar.ts

```diff
@@ -29,6 +30,15 @@ export const useSnackBar = () => {
     ({ set }) =>
       (newValue) =>
         set(snackBarInternalScopedState({ scopeId }), (prev) => {
+          if (
+            isDefined(newValue.dedupeKey) &&
```

Comment:
Ah, this makes sense now! I didn't realize the issue was rooted at the snackbar level. I've been banging my head trying to implement workarounds: avoiding useEffect, steering clear of useRef for state, and adding guards. Thanks for catching this!

Reply:
Yes sorry I think we led you in the wrong direction. Initial loading is probably the only case where we'll keep useEffects ; in that case we want this to be done at a high level (page level), not in sub-components.
Solving the double notification at the snackbar level was probably the cleanest way to make it more idempotent in strict mode, even though it's not perfect either